erpnext/accounts/doctype/asset/asset.py

```diff
@@ -80,7 +80,7 @@ def validate_asset_values(self):
 			if cint(self.number_of_depreciations_booked) > cint(self.total_number_of_depreciations):
 				frappe.throw(_("Number of Depreciations Booked cannot be greater than Total Number of Depreciations"))
 						
-		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(nowdate()):
+		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(self.purchase_date):
```

Comment:
If next depreciation date is before nowdate, that means you are creating an existing asset and its depreciation entry is not booked manually via Journal Entry. But we believe for most existing assets, their past depreciation entry should be booked via Journal Entry or in other accounting system (if user migrating to erpnext). There is a huge chance to enter a past date mistakenly. Thats why we added the warning message, maybe we should a "Warning" label before the message.

But we should surely add another validation for purchase date, it definitely can't be before purchase date.